packages/eslint-plugin/src/rules/ban-ts-comment.ts

```diff
@@ -0,0 +1,82 @@
+import * as util from '../util';
+
+interface Options {
+  'ts-ignore'?: boolean;
+  'ts-nocheck'?: boolean;
+  'ts-check'?: boolean;
+}
+
+const defaultOptions: [Options] = [
+  {
+    'ts-ignore': true,
+    'ts-nocheck': true,
+    'ts-check': false,
+  },
+];
+
+type MessageIds = 'tsDirectiveComment';
+
+export default util.createRule<[Options], MessageIds>({
+  name: 'ban-ts-comment',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Bans “// @ts-<directive>” comments from being used',
+      category: 'Best Practices',
+      recommended: 'error',
+    },
+    messages: {
+      tsDirectiveComment:
+        'Do not use "// @ts-{directive}" because it alters compilation errors.',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          'ts-ignore': {
+            type: 'boolean',
+            default: true,
+          },
+          'ts-nocheck': {
+            type: 'boolean',
+            default: true,
+          },
+          'ts-check': {
+            type: 'boolean',
+            default: false,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions,
+  create(context, [options]) {
+    const tsCommentRegExp = /^\/*\s*@ts-(ignore|check|nocheck)/;
+    const sourceCode = context.getSourceCode();
+
+    return {
+      Program(): void {
+        const comments = sourceCode.getAllComments();
+
+        comments.forEach(comment => {
+          if (comment.type !== 'Line') {
+            return;
+          }
+
+          const [, directive] = tsCommentRegExp.exec(comment.value) ?? [];
+
+          const fullDirective = `ts-${directive}` as keyof Options;
```

Comment:
I couldn't come up with a single nice way to handle this, so let me know how you'd prefer to do it.